packages/template-retail-react-app/app/utils/marketing-cloud/marketing-cloud-email-link.js

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * This file is responsible for integrating with the Marketing Cloud APIs to
+ * send emails with a magic link to a specified contact using the Marketing Cloud API.
+ * For this integration to work, a template email with a `%%magic-link%%` personalization string inserted
+ * must exist in your Marketing Cloud org.
+ * 
+ * More details here: https://developer.salesforce.com/docs/marketing/marketing-cloud/guide/transactional-messaging-get-started.html
+ *
+ * High Level Flow:
+ * 1. It retrieves an access token from the Marketing Cloud API using the
+ *    provided client ID and client secret.
+ * 2. It constructs the email message URL using the generated unique ID and the
+ *    provided template ID.
+ * 3. It sends the email message containing the magic link to the specified contact 
+ *    using the Marketing Cloud API.
+ */
+
+import crypto from 'crypto'
+import https from 'https'
+
+/**
+ * Tokens are valid for 20 minutes. We store it at the top level scope to reuse
+ * it during the lambda invocation. We'll refresh it after 15 minutes.
+ */
+let marketingCloudToken = ''
+let marketingCloudTokenExpiration = new Date()
+
+/*
+ Make a POST request using native https module, wrapped in a Promise with JSON
+ encode and decode
+*/
+export function asyncJsonHttpsPost(url, postObject, headers = {}) {
+    return new Promise((resolve, reject) => {
```

Comment:
Did you consider using async/await to match more of the recent code in the project? Or was this code something that was cut an pasted from some other lib?

Reply:
switched to use built-in fetch!